### PR TITLE
Intro infra override

### PR DESCRIPTION
**Pull request by jnpacker**
* Adjust comments
* Adjust requeues on error
* Add verb support for deletionCollection
* Add support for Spec.infrastructure.override  (INFRA-ONLY & ORPHAN)

### CODE REVIEW

File: pkg/controllers/hypershiftdeployment_controller.go [hunk 1]
```diff
@@ -87,6 +87,8 @@ func (r *HypershiftDeploymentReconciler) Reconcile(ctx context.Context, req ctrl
 		return ctrl.Result{}, nil
 	}
 
+	r.Log.Info("Reconciling")
```
Comment (ianzhang366) [inline pkg/controllers/hypershiftdeployment_controller.go:90]: can we do `log.Info("Reconciling")`, just to be consistent with the rest.

File: api/v1alpha1/hypershiftdeployment_types.go [hunk 2]
```diff
@@ -85,6 +90,11 @@ type InfraSpec struct {
 	// +immutable
 	Configure bool `json:"configure"`
 
+	// InfrastructureOverride allows support for special cases
+	//   InfraOverrideDestroy = "ORPHAN"
```
Comment (ianzhang366) [inline api/v1alpha1/hypershiftdeployment_types.go:94]: what do we mean or act on `ORPHAN`, can you please give me an example?

Comment (jnpacker) [reply api/v1alpha1/hypershiftdeployment_types.go:94]: Orphan means we delete the HypershiftDeployment resource and orphan the infrastructure.  So its in the infrastructureSpec for HD.

File: pkg/controllers/hypershiftdeployment_controller.go (hunk 1 above)
Comment (jnpacker) [reply pkg/controllers/hypershiftdeployment_controller.go:90]: Changed in this pull request https://github.com/stolostron/hypershift-deployment-controller/pull/10

File: api/v1alpha1/hypershiftdeployment_types.go (hunk 2 above)
Comment (jnpacker) [reply api/v1alpha1/hypershiftdeployment_types.go:94]: @ianzhang366 ^^

**Pull request merged**